pkg/kubectl/cmd/delete.go

```diff
@@ -248,13 +261,20 @@ func ReapResult(r *resource.Result, f cmdutil.Factory, out io.Writer, isDefaultD
 		if err != nil {
 			return err
 		}
+		// Set WaitForDeletion to false for namespace deletion unless it was explicitly set by the user.
+		if info.Mapping.Resource == "Namespace" && isDefaultWaitForDeletion {
```

Comment:
The fact that it is not uniform (some resources delete synchronously, some delete asynchronously) does not imply that kubectl polling/waiting should become the default. The fact that we'd have to special-case multiple resources not to wait for by default makes it very difficult to understand. If users opt into waiting, we don't need special cases.